src/main/java/baritone/pathing/movement/movements/MovementParkour.java

```diff
@@ -103,19 +105,35 @@ public static void cost(CalculationContext context, int x, int y, int z, EnumFac
             }
         }
         for (int i = 2; i <= maxJump; i++) {
-            // TODO perhaps dest.up(3) doesn't need to be fullyPassable, just canWalkThrough, possibly?
-            for (int y2 = 0; y2 < 4; y2++) {
-                if (!MovementHelper.fullyPassable(context, x + xDiff * i, y + y2, z + zDiff * i)) {
-                    return;
+            int destX = x + xDiff * i;
+            int destZ = z + zDiff * i;
+            if (!MovementHelper.fullyPassable(context, destX, y + 1, destZ)) {
+                return;
+            }
+            if (!MovementHelper.fullyPassable(context, destX, y + 2, destZ)) {
+                return;
+            }
+            if (!MovementHelper.fullyPassable(context, destX, y, destZ)) {
+                if (i <= 3 && context.allowParkourAscend && context.canSprint && MovementHelper.canWalkOn(context.bsi, destX, y, destZ) && checkOvershootSafety(context.bsi, destX + xDiff, y + 1, destZ + zDiff)) {
+                    res.x = destX;
+                    res.y = y + 1;
+                    res.z = destZ;
+                    res.cost = costFromJumpDistance(i) + context.jumpPenalty;
```

Comment:
costFromJumpDistance returns the cost of a standard jump. This needs to be changed because for example, when distance is 3, costFromJumpDistance(3) returns `WALK_ONE_BLOCK_COST * 3` while you really want it to be `SPRINT_ONE_BLOCK_COST * 3` since you sprint in the execution of this movement.

This line should instead be `res.cost = i * SPRINT_ONE_BLOCK_COST + context.jumpPenalty;`

Reply:
wow this is a serious bug